app/models/question.rb

```diff
@@ -1,2 +1,54 @@
 class Question < ActiveRecord::Base
+
+	validates_presence_of :content, :title
+	validates :title, length: {maximum: 150}
+	validates :content, length: {maximum: 5000}
+
+	belongs_to :submitter, class_name: :User
+	has_many :answers
+	has_many :comments, as: :commentable
+	has_many :votes, as: :voteable
+
+	def rating
+		votes.pluck(:value).reduce(:+) || 0
+	end
+
+	def has_best_answer?
+		best_answer_id
+	end	
+
+	def votes_per_hour
+		votes.select {|vote| vote.created_at < 1.hour.ago}.count
```

Comment:
Something like ```votes.where('created_at < ?', 1.hour.ago).count``` would work better. The way it is now it's going to load all votes then filter them.